corgidrp/photon_counting.py

```diff
@@ -284,17 +291,18 @@ def get_pc_mean(input_dataset, pc_master_dark=None, T_factor=None, pc_ecount_max
         if pc_master_dark is not None:
             if type(pc_master_dark) is not data.Dark:
                 raise Exception('Input type for pc_master_dark must be a Dataset of corgidrp.data.Dark instances.')
+            if (pc_master_dark.ext_hdr['EXPTIME'], pc_master_dark.ext_hdr['EMGAIN_C']) != unique_vals[0][:2]:
+                raise PhotonCountException('Dark should have the same EXPTIME and EMGAIN_C as input_dataset.')
             if 'PC_STAT' not in pc_master_dark.ext_hdr:
                 raise PhotonCountException('\'PC_STAT\' must be a key in the extension header of each frame of pc_master_dark.')
-            if pc_master_dark.ext_hdr['PC_STAT'] != 'photon-counted master dark':
-                raise PhotonCountException('Each frame of pc_master_dark must be a photon-counted master dark (i.e., '
-                                        'the extension header key \'PC_STAT\' must be \'photon-counted master dark\').')
-            if 'PCTHRESH' not in pc_master_dark.ext_hdr:
-                raise PhotonCountException('Threshold should be stored under the header \'PCTHRESH\'.')
-            if pc_master_dark.ext_hdr['PCTHRESH'] != thresh:
-                raise PhotonCountException('Threshold used for photon-counted master dark should match the threshold to be used for the illuminated frames.')
-            if pc_master_dark.ext_hdr['NUM_FR'] < len(sub_dataset):
-                raise PhotonCountException('Number of frames that created the photon-counted master dark must be greater than or equal to the number of illuminated frames in order for the result to be reliable.')
+            if pc_master_dark.ext_hdr['PC_STAT'] == 'photon-counted master dark':
+                if 'PCTHRESH' not in pc_master_dark.ext_hdr:
+                    raise PhotonCountException('Threshold should be stored under the header \'PCTHRESH\'.')
+                if pc_master_dark.ext_hdr['PCTHRESH'] != thresh:
+                    raise PhotonCountException('Threshold used for photon-counted master dark should match the threshold to be used for the illuminated frames.')
+                if pc_master_dark.ext_hdr['NUM_FR'] < len(sub_dataset):
+                    raise Warning('Number of frames that created the photon-counted master dark should be greater than or equal to the number of illuminated frames in order for the result to be reliable.')
```

Comment:
do you mean warnings.warn? 

I'm also thinking whether this should be a print statement instead of a warning.

Reply:
yes, unless this is specifically important enough for a warning (maybe not)